src/dodal/devices/undulator.py

```diff
@@ -23,12 +23,28 @@ class Undulator(StandardReadable):
     def __init__(
         self,
         prefix: str,
+        poles: int,
+        length: float,
```

Comment:
@DominicOram Interested in your thoughts about requiring this information in the constructor before I go ahead and find it out for all beamlines. It's a bit forceful but does guarantee that all `Undulator` instances carry enough information for making an `NXInsertionDevice`

Reply:
I'm happy for this to be required information but it's not required for a NXInsertionDevice to be written, both those fields are optional in https://manual.nexusformat.org/classes/base_classes/NXinsertion_device.html#nxinsertion-device

Reply:
That's a good point! It would be ideal if it was optional then, and the value only made it into the `read` output if a user supplied it. Not sure how ophyd-async deals with optional signals/if it should. @coretl might have some comments.

Reply:
@coretl says that similar things in panda are optional signals (i.e. the attribute is just `None` if no value supplied)